internal/image/build.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package image
+
+import (
+	"fmt"
+	"net/mail"
+	"regexp"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/storage/memory"
+)
+
+var gitClone = git.Clone
+
+type invocation struct {
+	ConfigSource map[string]interface{} `json:"configSource"`
+	Parameters   map[string]string      `json:"parameters"`
+	Environment  map[string]interface{} `json:"environment"`
+}
+
+type materials struct {
+	Uri    string            `json:"sha"`
+	Digest map[string]string `json:"digest"`
+}
+
+type predicate struct {
```

Comment:
Wouldn't this be sufficient?
```go
type predicate struct {
	Materials   []materials            `json:"materials"`
}
```

Same for `attestation`. I think we can remove the fields we don't care about.